src/Features/Core/Portable/IntroduceVariable/AbstractIntroduceParameterService.cs

```diff
@@ -0,0 +1,656 @@
+        }
+
+        /// <summary>
+        /// Generates a method declaration containing a return expression of the highlighted expression
+        /// </summary>
+        private async Task IntroduceNewMethodAsync(SemanticDocument semanticDocument,
+                                                   TExpressionSyntax expression,
+                                                   IMethodSymbol methodSymbol,
+                                                   string newMethodIdentifier,
+                                                   SyntaxEditor editor,
+                                                   CancellationToken cancellationToken)
+        {
+            var document = semanticDocument.Document;
+            var syntaxFacts = semanticDocument.Document.GetRequiredLanguageService<ISyntaxFactsService>();
+            var generator = SyntaxGenerator.GetGenerator(document);
+            var root = await document.GetRequiredSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            var methodExpression = expression.FirstAncestorOrSelf<SyntaxNode>(node => IsMethodDeclaration(node), true)!;
+            var returnExpression = new List<SyntaxNode>
+            {
+                generator.ReturnStatement(expression.WithoutTrailingTrivia())
+            };
+
+            var typeSymbol = semanticDocument.SemanticModel.GetTypeInfo(expression, cancellationToken).Type;
+            var newMethod = CodeGenerationSymbolFactory.CreateMethodSymbol(methodSymbol, name: newMethodIdentifier, returnType: typeSymbol);
+
+            var methodDeclaration = generator.MethodDeclaration(newMethod, returnExpression).WithAdditionalAnnotations(Formatter.Annotation);
+            editor.InsertBefore(methodExpression, methodDeclaration);
+        }
+
+        /// <summary>
+        /// Generates a method declaration containing a call to the method that introduced the parameter
+        /// </summary>
+        /// 
+        private async Task IntroduceNewMethodOverloadAsync(SemanticDocument semanticDocument,
+                                                           TExpressionSyntax expression,
+                                                           IMethodSymbol methodSymbol,
+                                                           SyntaxEditor editor,
+                                                           CancellationToken cancellationToken)
+        {
+            var document = semanticDocument.Document;
+            var syntaxFacts = semanticDocument.Document.GetRequiredLanguageService<ISyntaxFactsService>();
+            var generator = SyntaxGenerator.GetGenerator(document);
+            var root = await document.GetRequiredSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            var methodExpression = expression.FirstAncestorOrSelf<SyntaxNode>(node => IsMethodDeclaration(node), true)!;
+            var arguments = generator.CreateArguments(methodSymbol.Parameters.As<IParameterSymbol>());
```

Comment:
i don't understand the .As part her.   a methodSymbol.Parameters is already a list of IParameterSymbols

Reply:
I was just making sure the type was an IParameterSymbol, removed since it's unnecessary.